services/web/server/src/simcore_service_webserver/notifications/_rabbitmq_nonexclusive_queue_consumers.py

```diff
@@ -0,0 +1,102 @@
+import functools
+import logging
+from collections.abc import AsyncIterator, Callable, Coroutine
+from typing import Any, Final, NamedTuple
+
+from aiohttp import web
+from models_library.rabbitmq_messages import InstrumentationRabbitMessage
+from servicelib.aiohttp.monitor_services import (
+    MONITOR_SERVICE_STARTED_LABELS,
+    MONITOR_SERVICE_STOPPED_LABELS,
+    service_started,
+    service_stopped,
+)
+from servicelib.logging_utils import log_catch, log_context
+from servicelib.rabbitmq import RabbitMQClient
+from servicelib.utils import logged_gather
+
+from ..rabbitmq import get_rabbitmq_client
+from ._constants import APP_RABBITMQ_CONSUMERS_KEY
+
+_logger = logging.getLogger(__name__)
+
+
+async def _instrumentation_message_parser(app: web.Application, data: bytes) -> bool:
+    rabbit_message = InstrumentationRabbitMessage.parse_raw(data)
+    if rabbit_message.metrics == "service_started":
+        service_started(
+            app,
+            **{
+                key: rabbit_message.dict()[key]
+                for key in MONITOR_SERVICE_STARTED_LABELS
+            },
+        )
+    elif rabbit_message.metrics == "service_stopped":
+        service_stopped(
+            app,
+            **{
+                key: rabbit_message.dict()[key]
+                for key in MONITOR_SERVICE_STOPPED_LABELS
+            },
+        )
+    return True
```

Comment:
Thought: I always found strange to have functions that just return `True`. When do you return `False`? Should you perhaps catch exceptions and return False in that case? e.g. a `KeyError`?

Reply:
Yes, rabbitmq expects True so it knows that the message for successfully consumed, if it fails on the way because of some error it will return false automatically.